### PR TITLE
make connection_type enum an enum class

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

File: include/libtorrent/peer_connection_handle.hpp [hunk 1]
```diff
@@ -57,7 +58,7 @@ struct TORRENT_EXPORT peer_connection_handle
 		: m_connection(impl)
 	{}
 
-	int type() const;
+	connection_type type() const;
```
Comment (aldenml) [inline include/libtorrent/peer_connection_handle.hpp]: I think this could be a problem since `connection_type` is a private type

Comment (arvidn) [inline include/libtorrent/peer_connection_handle.hpp]: hm.. maybe it shouldn't be..

Comment (arvidn) [inline include/libtorrent/peer_connection_handle.hpp]: actually, technically it is public. it's not in the ``aux`` namespace nor in the ``aux_`` directory. Once ``peer_connection.hpp`` gets moved, this type should remain though

Comment (aldenml) [inline include/libtorrent/peer_connection_handle.hpp]: now I need to tweak the swig file :)

**Pull request merged**